test/addons/async-hooks-promise/test.js

```diff
@@ -5,6 +5,11 @@ const assert = require('assert');
 const async_hooks = require('async_hooks');
 const binding = require(`./build/${common.buildType}/binding`);
 
+if (process.env.NODE_TEST_WITH_ASYNC_HOOKS) {
```

Comment:
What are the tradeoffs of this VS doing `delete process.env.NODE_TEST_WITH_ASYNC_HOOKS` before L3

Reply:
Not sure what you mean. Mind clarifying?

Reply:
If we delete `NODE_TEST_WITH_ASYNC_HOOKS` before the L3 (`const common = require('../../common');`) it will not run the hooking logic of the harness https://github.com/nodejs/node/blob/b2865ba8a3330f763e8a3833a82bef24660de916/test/common/index.js#L64-L65

Will that make this testable?

Reply:
while the possibility does exist, we depend on env vars existing for other things that could also be deleted. So I'm not going to worry about it.